extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -57,9 +96,81 @@ export class AmpSlideScroll extends BaseCarousel {
 
   /** @override */
   hasPrev() {
+    return this.slideIndex_ > 0;
   }
 
   /** @override */
   hasNext() {
+    return this.slideIndex_ < this.slides_.length - 1;
+  }
+
+  /** @override */
+  goCallback(dir, unusedAnimate) {
+    if (this.slideIndex_ != null) {
+      if ((dir == DIR_FORWARD && this.hasNext()) ||
+          (dir == DIR_REVERSE && this.hasPrev())) {
+        this.mutateElement(() => {
+          this.showSlides_(this.slideIndex_ + dir);
+        }, this.slidesContainer_);
+      }
+    }
+  }
+
+  /**
+   * Makes the slide corresponding to the given index and the slides surrounding
+   *    it available for display.
+   * @param {number} newindex Index of the slide to be displayed.
+   * @private
+   */
+  showSlides_(newindex) {
+    const noOfSlides = this.noOfSlides_;
+    if (newindex < 0 ||
+        newindex >= this.noOfSlides_ ||
+        this.slideIndex_ == newindex) {
+      return;
+    }
+    const showIndexArr = [];
+    let newScrollLeft;
+    if (newindex == noOfSlides - 1) {
+      // Last slide.
+      showIndexArr.push(noOfSlides - 1, noOfSlides - 2);
+    } else if (newindex === 0) {
+      // First slide.
+      showIndexArr.push(0, 1);
+      newScrollLeft = 0;
+    } else {
+      showIndexArr.push(newindex - 1, newindex, newindex + 1);
+    }
+
+    if (newScrollLeft == null) {
+      newScrollLeft = this.slideWidth_;
+    }
+    if (this.slideIndex_ != null) {
+      this.updateInViewport(this.slides_[this.slideIndex_], false);
+    }
+    this.updateInViewport(this.slides_[newindex], true);
+    this.slideIndex_ = newindex;
+    showIndexArr.forEach(showIndex => {
+      this.slideWrappers_[showIndex].setAttribute('show', '');
+      this.scheduleLayout(this.slides_[showIndex]);
+    });
+    this.slidesContainer_./*REVIEW*/scrollLeft = newScrollLeft;
+    this.hideRestOfTheSlides_(newindex);
+    this.setControlsState();
+  }
+
+  /**
+   * Given an index, hides rest of the slides that are not needed.
+   * @param {number} index Index of the slide to be displayed.
+   * @private
+   */
+  hideRestOfTheSlides_(index) {
+    for (let i = 0; i < this.noOfSlides_; i++) {
+      if (i != index && i != index - 1 && i != index + 1 &&
+          this.slideWrappers_[i]) {
+        this.slideWrappers_[i].removeAttribute('show');
+        this.schedulePause(this.slides_[i]);
```

Comment:
Fixed it here - Follow up question to @dvoytenko 
This means that i will have to maintain an array of slide states. I thought if we called schedulePause on an element that was not laid out it did nothing. Shouldn't this be something the API itself should do?